### DISCUSSION TITLE
How to set the geometry attribute and dimtype attribute of the DIMENSION entity of EZDXF？

### DISCUSSION THREAD

**Discussion by luoyu-123** [mozman/ezdxf#274] [Q&A]
![image](https://user-images.githubusercontent.com/59160360/101436090-70555080-3948-11eb-88dc-d9c6b08232ed.png)

I want to change the geometric properties  of the DIMENSION entity, but my changes in the dxfattribs method do not take effect

![image](https://user-images.githubusercontent.com/59160360/101436674-99c2ac00-3949-11eb-8d5b-f142dd559181.png)

**Comment by mozman** [comment]
The `geometry` attribute is the handle to the anonymous block which contains all the dimension geometry as DXF primitives.
Why do you want to change that?

**Comment by mozman** [comment]
The name structure for anonymous dimension blocks has to be: `"*D<unique decimal number>"`.

**Comment by luoyu-123** [comment]
That is to say, if I want to create a linear size style by myself, can I only create anonymous blocks? 
such as：
![image](https://user-images.githubusercontent.com/59160360/101453969-2a5db400-396b-11eb-9820-f0714a26fc6c.png)

**Comment by luoyu-123** [comment]
Because i didn't find the relevant style

**Comment by mozman** [comment]
To be clear, ezdxf is not a CAD application and creating fancy annotations, dimensions and text entities was never the intended purpose of this package. YES to archive effects like in the previous comment you have to render the DIMENSION line by yourself into an anonymous block as DXF primitives like LINE, CIRCLE, INSERT and TEXT and there is no tutorial for that.

My opinion to auto-generated annotations and dimension as professional CAD engineer: most of the time they are ugly, misplaced and useless, I would never waste my time for creating them.

**Comment by luoyu-123** [comment]
thanks
